### ISSUE TITLE
Floor Ceiling

### ISSUE THREAD

**Issue by chrisbach1** [DimaKudosh/pydfs-lineup-optimizer#399]
I would love some input here.

I have added Projection Floor and Projection Ceiling columns to my csv file.  Do I still need to add the RandomFantasyPointStrategy to my code?  How do I get the optimizer to project fppg in the range?

Anything helps!  Thanks in advance.  Love this optimizer

**Comment by Denwen12** [comment]
RandomFantasyPointsStrategy needs to be added

**Comment by chrisbach1** [comment]
What do I put as the range since I am using floor and ceiling columns in the csv?

**Comment by Denwen12** [comment]
You can usually can find a players floor and ceiling online

**Comment by Denwen12** [comment]
You can look on dk and fd and look at the players details and find his
floor and his highest Fantasy points games and use that

**Comment by chrisbach1** [comment]
I mean how do I get the optimizer to randomize a projection between my floor and ceiling.  Will it do it automatically if I add the RandomFantasyPointStrategy?

**Comment by chrisbach1** [comment]
I have the columns and numbers already

**Comment by Milkman9691** [comment]
Floors & Ceilings you find online are useless unless they give you every detail as to how they created them. It is pretty simple to create your own and then you know if players are mixed in with only 5 mins avg playing time but I would not use any found online NBC Sports Edge has a good write up about creation of a F&C chart so i would start there

**Comment by Denwen12** [comment]
Yes when you run the optimizer you'll see fantasy pou ts in  Parenthesis

**Comment by chrisbach1** [comment]
So if I am specifying my floors and ceilings I just add this?:

optimizer.set_fantasy_points_strategy(RandomFantasyPointsStrategy())


I appreciate the help

**Comment by Denwen12** [comment]
Should work fine

**Comment by chrisbach1** [comment]
@Denwen12 Thanks!

**Comment by Denwen12** [comment]
Did you import the function?

**Comment by Denwen12** [comment]
No problem

**Comment by chrisbach1** [comment]
@Denwen12 so Im still having trouble with it.  My understanding is that if I add optimizer.set_fantasy_points_strategy(RandomFantasyPointsStrategy())  along with the floor & ceiling categories to my csv, that the optimizer would assign a random number between the floor and ceiling.  Essentially  assigning a player a random FPPG within the floor and ceiling.     But if I keep the floor and ceiling in the csv and change all the players FPPG to 0,  it does not assign any number.   does that make sense? @DimaKudosh

**Comment by Denwen12** [comment]
Send a screenshot if your csv

**Comment by chrisbach1** [comment]
![Screenshot 2023-01-09 170334](https://user-images.githubusercontent.com/89428871/211425798-42baf8d4-8863-4d41-8b63-ad078eef1aee.png)

**Comment by chrisbach1** [comment]
Okay its supposed to be "Projection Ceil", not "Projection Ceiling"    That fixed it

**Comment by Denwen12** [comment]
Your columns are named wrong

   - Max Exposure
   - Min Exposure
   - Roster Order
   - Projected Ownership
   - Min Deviation
   - Max Deviation
   - Projection Floor
   - Projection Ceil
   - Confirmed Starter
   - Progressive Scale

**Comment by Denwen12** [comment]
Your ownership is wrong also
